controllers/observability/perses.go

```diff
@@ -0,0 +1,215 @@
+package observability
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	authv1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	persesv1alpha1 "github.com/rhobs/perses-operator/api/v1alpha1"
+
+	assets "github.com/kubevirt/hyperconverged-cluster-operator/assets"
+)
+
+const (
+	persesDatasourceSecretName = "perses-thanos-datasource-secret"
+)
+
+// persesAssetsDir is the directory (inside the repo image) containing Perses CR manifests (yaml).
+// Each YAML should define a single resource: PersesDashboard or PersesDatasource.
+const persesAssetsDir = "."
+
+// ReconcilePersesResources applies all Perses resources shipped with the operator under assets/perses.
+// It uses server-side apply to create/update resources idempotently without relying on cache reads.
+func (r *Reconciler) ReconcilePersesResources(ctx context.Context) error {
+	// Use embedded assets instead of runtime file system
+	d := assets.GetPersesDashboardsFS()
+
+	// Ensure Perses CRDs are present before attempting to apply resources.
+	// Without CRDs, server-side apply will fail on every reconcile loop.
+	if !r.persesCRDsReady(ctx) {
+		return nil
+	}
+
+	// Load and cache assets once
+	var loadErr error
+	r.assetsOnce.Do(func() {
+		r.cachedAssets, loadErr = readPersesAssets(d)
+	})
+	if loadErr != nil {
+		return loadErr
+	}
```

Comment:
This error is not recoverable. On the other hand, if the asset files are wrong, then we should not pass unit tests.

I think it will be safe to panic here, because if our unit tests are good, then we shall never get error here, in production.

WDYT about this?

```suggestion
	r.assetsOnce.Do(func() {
		var err error
		r.cachedAssets, err = readPersesAssets(d)
		if err != nil {
			panic("failed to parse perses files; " + err.Error())
		}
	})
```